File: lib/widget_constants.dart
/// Shared preference keys and defaults (mirror [WidgetConstants.kt]).
abstract final class WidgetConstants {
  static const PREFS_NAME = 'nl.siwoc.calendarwidget';

  static const KEY_CALENDAR_WIDGET_DATA = 'calendar_widget_data';

  static const KEY_HEADER_COLOR = 'header_color';
  static const KEY_HEADER_FONT_SIZE = 'header_font_size';
  static const KEY_EVENT_FONT_SIZE = 'event_font_size';
  static const KEY_FETCH_DAYS = 'fetch_days';
  static const KEY_LOCALE = 'locale';
  static const KEY_SELECTED_CALENDAR_IDS = 'selected_calendar_ids';
  static const KEY_BACKGROUND_COLOR = 'background_color';
  static const KEY_BACKGROUND_OPACITY = 'background_opacity';

  static const METHOD_CHANNEL = 'nl.siwoc.calendarwidget/calendar';

  static const DEFAULT_HEADER_COLOR = '#FF000000';
  static const DEFAULT_HEADER_FONT_SIZE = 14;
  static const DEFAULT_EVENT_FONT_SIZE = 14;
  static const DEFAULT_FETCH_DAYS = 7;
  static const DEFAULT_LOCALE = 'nl-NL';
  static const DEFAULT_BACKGROUND_COLOR = '#4A4A4A4A';
  static const DEFAULT_BACKGROUND_OPACITY = 50;

  /// [CalendarWidgetData] JSON schema version.
  static const SCHEMA_VERSION = 1;

  /// [CalendarWidgetError.code] values written by the refresh worker.
  static const ERROR_PERMISSION_DENIED = 'permission_denied';
  static const ERROR_NO_CALENDARS = 'no_calendars';
  static const ERROR_UNKNOWN = 'unknown';
}

File: android/app/src/main/kotlin/nl/siwoc/calendarwidget/WidgetConstants.kt
package nl.siwoc.calendarwidget

object WidgetConstants {
    const val PREFS_NAME = "nl.siwoc.calendarwidget"

    /** Snapshot JSON written by the worker; read by Glance and Flutter preview. */
    const val KEY_CALENDAR_WIDGET_DATA = "calendar_widget_data"

    const val KEY_HEADER_COLOR = "header_color"
    const val KEY_HEADER_FONT_SIZE = "header_font_size"
    const val KEY_EVENT_FONT_SIZE = "event_font_size"
    const val KEY_FETCH_DAYS = "fetch_days"
    const val KEY_LOCALE = "locale"
    const val KEY_SELECTED_CALENDAR_IDS = "selected_calendar_ids"
    const val KEY_BACKGROUND_COLOR = "background_color"
    const val KEY_BACKGROUND_OPACITY = "background_opacity"

    const val METHOD_CHANNEL = "nl.siwoc.calendarwidget/calendar"

    /** Unique name for [CalendarRefreshScheduler] periodic WorkManager job. */
    const val PERIODIC_REFRESH_WORK_NAME = "calendar_widget_periodic_refresh"

    /** Background calendar refresh interval (minutes). Android minimum is 15. */
    const val PERIODIC_REFRESH_INTERVAL_MINUTES = 30L

    // Readable hex; opaque black.
    const val DEFAULT_HEADER_COLOR = "#FF000000"
    const val DEFAULT_HEADER_FONT_SIZE = 14
    const val DEFAULT_EVENT_FONT_SIZE = 14
    const val DEFAULT_FETCH_DAYS = 7
    const val DEFAULT_LOCALE = "nl-NL"
    const val DEFAULT_BACKGROUND_COLOR = "#4A4A4A4A"
    const val DEFAULT_BACKGROUND_OPACITY = 30

    /** [CalendarWidgetData] JSON schema version. */
    const val SCHEMA_VERSION = 1

    /** [CalendarWidgetError.code] values written by the refresh worker. */
    const val ERROR_PERMISSION_DENIED = "permission_denied"
    const val ERROR_NO_CALENDARS = "no_calendars"
    const val ERROR_UNKNOWN = "unknown"
}
